scripts/check_e2e_tests_are_captured_in_ci.py

```diff
@@ -0,0 +1,148 @@
+
+    Returns:
+        list(str). A list of names of test suites in sorted alphabetical
+        order of script section in travis.yml file.
+    """
+    travis_file_content = read_and_parse_travis_yml_file()
+    script_str = python_utils.convert_to_bytes(travis_file_content['script'])
+    # The following line extracts the test suites from patterns like
+    # bash scripts/run_e2e_tests.sh --suite="accessibility"
+    hyphen_between_regex = re.compile(
+        r'bash scripts/run_e2e_tests.sh --suite="([a-zA-Z_-]*)"')
+    suites_list = hyphen_between_regex.findall(script_str)
+
+    return sorted(suites_list)
+
+
+def get_e2e_suite_names_from_protractor_file():
+    """Extracts the test suites section from the .travis.yml file.
+
+    Returns:
+        list(str). A list of all test suites in sorted order
+        in protractor.conf.js file.
+    """
+    protractor_config_file_content = read_protractor_conf_file()
+    # The following line extracts suite object from protractor.conf.js.
+    suite_object_string = re.compile(
+        r'suites = {([^}]+)}').findall(protractor_config_file_content)[0]
+
+    # The following line extracts the keys/test suites from the key: value
+    # pair from the suites object.
+    key_regex = re.compile(r'\b([a-zA-Z_-]*)(?=:)')
+    protractor_suites = []
+    for match in key_regex.finditer(suite_object_string):
+        if match.group():
+            protractor_suites.append(match.group())
+
+    return sorted(protractor_suites)
+
+
+def read_protractor_conf_file():
+    """Returns the contents of core/tests/protractor.conf.js file.
+
+    Returns:
+        str. The contents of protractor.conf.js, as a string.
+    """
+    protractor_config_file_content = python_utils.open_file(
+        os.path.join(
+            os.getcwd(), 'core', 'tests', 'protractor.conf.js'), 'r').read()
+    return protractor_config_file_content
+
+
+def read_and_parse_travis_yml_file():
+    """Returns the contents of .travis.yml, as a dict.
+
+    Returns:
+        dict. Contents of the travis.yml file parsed as a dict.
+    """
+    travis_ci_file_content = python_utils.open_file(
+        os.path.join(os.getcwd(), '.travis.yml'), 'r').read()
+    travis_ci_dict = utils.dict_from_yaml(travis_ci_file_content)
+    return travis_ci_dict
+
+
+def main():
+    """Test the travis ci file and protractor.conf.js to have same
+    e2e test suites.
+    """
+    python_utils.PRINT('Checking e2e tests are captured in travis.yml started')
+    protractor_test_suites = get_e2e_suite_names_from_protractor_file()
+    yaml_jobs = get_e2e_suite_names_from_jobs_travis_yml_file()
+    yaml_scripts = get_e2e_suite_names_from_script_travis_yml_file()
+
+    for excluded_test in TEST_SUITES_NOT_RUN_ON_TRAVIS:
+        protractor_test_suites.remove(excluded_test)
+
+    if not (len(yaml_jobs) > 0 and len(yaml_scripts) > 0
+            and len(protractor_test_suites) > 0):
+        raise Exception('The tests suites that have been extracted from '
```

Comment:
tests --> test

Can you break this down and say which one? That would help developers debug errors.

Reply:
Done